org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/AbstractProjectsManagerBasedTest.java

```diff
@@ -137,6 +143,22 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
 	public static void initJVMs() {
 		JobHelpers.waitForLookupJDKToolchainsJob(600000);
 		CorePlugin.publishedGradleVersions().getVersions();
+		// see https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/3261
+		IVMInstallType[] vmInstallTypes = JavaRuntime.getVMInstallTypes();
```

Comment:
Why is this necessary during the test runtime ? How is the failure occuring ?

Reply:
I have also replaced
```
assertEquals(lambda.getLabel(), "(Object arg0) ->");
```
with
```
assertEquals("(Object arg0) ->", lambda.getLabel());
```
at https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/0949cc2c6f558a63bc1a6735cea44c3f3972d1be/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerTest.java#L3464

Reply:
Before we get here, the VMs are already configured though. Is it done by `DetectVMInstallationsJob` ? would it not be better to call that again ?

Reply:
DetectVMInstallJob doesn't configure Java LS Test VMs. See 
- https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/f9425d878f596f88c8d64ae53b8be03da8d19bc4/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerTestPlugin.java#L60
- https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/f9425d878f596f88c8d64ae53b8be03da8d19bc4/org.eclipse.jdt.ls.tests/plugin.xml#L8
In addition, we run tests with -DDetectVMInstallationsJob.disabled=true
See https://ci.eclipse.org/ls/job/jdt-ls-pr/configure